src/foam/nanos/theme/Theme.js

```diff
@@ -584,5 +592,125 @@ foam.CLASS({
         return foam.util.SafetyUtil.isEmpty(getName()) || foam.util.SafetyUtil.isEmpty(getDescription()) ? "" : getName() + " " + getDescription();
       `
     },
+    {
+      name: 'merge',
+      type: 'Theme',
+      args: [ 'Theme other' ],
+      code: function(other) {
```

Comment:
you are copying over a theme(other) onto this theme. So 'other' is taking president. Hmm so in your case you have spid.theme and group.theme and parentGroup.themes. Your code runs with -`theme.merge(groupTheme)` - where theme is spid.theme and groupTheme is group.theme.
So I have a suggestion. 
If you make a theme object from all the group themes (collectiveGroupSpid). Also making the child themes always take precedent. So example) treviso-fraudOps theme only copies fields from fraud-ops theme iff treviso-fraudOps does not have that property set and fraud-ops.theme does.

Then copy the spid.theme to the collectiveGroupTheme, making collectiveGroupSpid take precedent.

Reply:
IMO, the collectiveGroupSpid theme could make it harder to troubleshoot as we would need to look at all the themes that constitute the final collective theme to know where a property setting is coming from. Eg, if all ancestor groups and the group the user belong to all have associated theme then we would have find which theme among them is responsible for a particular styling we wanted to fix.

Where as for the current approach, we only have to look at the first group theme and the spid theme.

Reply:
hmmm groups are hierarchal - but you don't want the theme on spid to be hierarchal. I get your concern about finding the right configuration.

Reply:
I'm good with how it stands, but if we were to consider group hierarchy, then it would be copyFrom/merge(parentGroup), then copyFrom/merge(ownGroup).  So recurse through the parent groups then copyFrom/merge from each on return.  
Is this what you wanted @Nauna